Sources/HiPMobile/DataAccessLayer/DbContextDebugView.cs

```diff
@@ -0,0 +1,284 @@
+    }
+
+    [DebuggerDisplay("{Entity}")]
+    public class EntityDebugView
+    {
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private readonly DbContext db;
+
+        public EntityDebugView(DbContext db, object entity)
+        {
+            this.db = db ?? throw new ArgumentNullException(nameof(db));
+            Entity = entity;
+        }
+
+        // Note: 'DebuggerBrowsableState.HideRoot' causes 'References' to be missing in Xamarin
+        public object Entity { get; set; }
+
+        public NavigationCollectionDebugView References => new NavigationCollectionDebugView(db, Entity);
+    }
+
+    /// <summary>
+    /// Lists the navigation properties of an entity and loads them on demand.
+    /// </summary>
+    [DebuggerDisplay("🔗 References")]
+    public class NavigationCollectionDebugView
+    {
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private readonly DbContext db;
+
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private readonly object entity;
+
+        public NavigationCollectionDebugView(DbContext db, object entity)
+        {
+            this.db = db ?? throw new ArgumentNullException(nameof(db));
+            this.entity = entity;
+        }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+        public IReadOnlyList<NavigationDebugView> References => (entity == null)
+            ? ImmutableList<NavigationDebugView>.Empty
+            : db.Entry(entity).Navigations
+                .Select(nav => new NavigationDebugView(db, nav))
+                .ToList() as IReadOnlyList<NavigationDebugView>;
+    }
+
+    /// <summary>
+    /// Displays a navigation property and loads it on demand.
+    /// </summary>
+    [DebuggerDisplay("{Description}")]
+    public class NavigationDebugView
+    {
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private readonly DbContext db;
+
+        [DebuggerBrowsable(DebuggerBrowsableState.Never)]
+        private readonly NavigationEntry nav;
+
+        public NavigationDebugView(DbContext db, NavigationEntry nav)
+        {
+            this.db = db ?? throw new ArgumentNullException(nameof(db));
+            this.nav = nav ?? throw new ArgumentNullException(nameof(nav));
+        }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+        public object Value
+        {
+            get
+            {
+                // For nav.Query() we need to ensure that the entity is (at least temporarily) attached
+                var isDetached = nav.EntityEntry.State == EntityState.Detached;
+
+                if (isDetached)
+                    nav.EntityEntry.State = EntityState.Unchanged;
+
+                try
+                {
+                    switch (nav)
+                    {
+                        // Note: Query() won't attach to change tracker
+
+                        // TODO: This doesn't seem to work correctly
```

Comment:
Should this be merged then?

Reply:
Since this is not impacting app functionality in any way and I couldn't find a robust solution yet, I think we don't need to fix this now. Remember, it's only for debugging purposes - let's just write bug-free code, then we don't even need this 😄